src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/KerberosReceiverSecurityToken.cs

```diff
@@ -0,0 +1,135 @@
+                                byte[] request, 
+                                string id, 
+                                bool doAuthenticate, 
+                                string valueTypeUri, 
+                                ChannelBinding channelBinding, 
+                                ExtendedProtectionPolicy extendedProtectionPolicy )
+        {
+            if (request == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentNullException(nameof(request)));
+            if (id == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentNullException(nameof(id)));
+
+            _id = id;
+            _request = request;
+            this.ValueTypeUri = valueTypeUri;
+            this._channelBinding = channelBinding;
+            this._extendedProtectionPolicy = extendedProtectionPolicy;
+
+            if (doAuthenticate)
+            {
+                Initialize(null, channelBinding, extendedProtectionPolicy);
+            }
+        }
+
+        public override ReadOnlyCollection<SecurityKey> SecurityKeys
+        {
+            get
+            {
+                if (_securityKeys == null)
+                {
+                    List<SecurityKey> temp = new List<SecurityKey>(1);
+                    temp.Add(SecurityKey);
+                    _securityKeys = temp.AsReadOnly();
+                }
+                return _securityKeys;
+            }
+        }
+
+        public SymmetricSecurityKey SecurityKey
+        {
+            get 
+            {
+                if (!_isAuthenticated)
+                {
+                    Initialize(null, _channelBinding, _extendedProtectionPolicy);
+                }
+                return _symmetricSecurityKey; 
+            }
+        }
+
+        public override DateTime ValidFrom
+        {
+            get
+            {
+                if (!_isAuthenticated)
+                {
+                    Initialize(null, _channelBinding, _extendedProtectionPolicy);
+                }
+                return base.ValidFrom;
+            }
+        }
+
+        public override DateTime ValidTo
+        {
+            get
+            {
+                if (!_isAuthenticated)
+                {
+                    Initialize(null, _channelBinding, _extendedProtectionPolicy);
+                }
+                return base.ValidTo;
+            }
+        }
+
+        public override WindowsIdentity WindowsIdentity
+        {
+            get
+            {
+                ThrowIfDisposed();
+                if (!_isAuthenticated)
+                {
+                    Initialize(null, _channelBinding, _extendedProtectionPolicy);
+                }
+                return base.WindowsIdentity;
+            }
+        }
+
+        private void Initialize(object p, ChannelBinding channelBinding, ExtendedProtectionPolicy extendedProtectionPolicy) => throw new NotImplementedException();
+
```

Comment:
You should be able to use the same implementation as you did for the Windows auth implementation for TransportWithMessageCredentials. Is there a reason why this didn't get implemented?